test/e2e/framework/framework.go

```diff
@@ -190,6 +192,18 @@ func (f *Framework) BeforeEach() {
 		f.StagingClient, err = staging.NewForConfig(clientRepoConfig)
 		Expect(err).NotTo(HaveOccurred())
 		f.ClientPool = dynamic.NewClientPool(config, api.Registry.RESTMapper(), dynamic.LegacyAPIPathResolverFunc)
+		if ProviderIs("kubemark") && TestContext.ExternalKubeConfig != "" && TestContext.CloudConfig.KubemarkProvider == nil {
+			kubemarkClient, err := clientset.NewForConfig(config)
+			Expect(err).NotTo(HaveOccurred())
+			externalConfig, err := clientcmd.BuildConfigFromFlags("", TestContext.ExternalKubeConfig)
+			externalConfig.QPS = f.Options.ClientQPS
+			externalConfig.Burst = f.Options.ClientBurst
+			Expect(err).NotTo(HaveOccurred())
+			externalClient, err := clientset.NewForConfig(externalConfig)
+			Expect(err).NotTo(HaveOccurred())
+			TestContext.CloudConfig.KubemarkProvider, err = kubemark.NewProvider(externalClient, kubemarkClient, make(chan struct{}))
```

Comment:
This channel doesn't serve any purpose. To make it meaningful, you should have the channel stored as a field inside `Framework` that you set here and later close inside `AfterEach()` method. For e.g - https://github.com/kubernetes/kubernetes/blob/master/test/e2e/framework/framework.go#L84